File: src/main/java/no/nav/bidrag/dokument/arkiv/security/SaksbehandlerInfoManager.java
package no.nav.bidrag.dokument.arkiv.security;

import java.util.Optional;
import no.nav.bidrag.commons.security.service.OidcTokenManager;
import no.nav.bidrag.dokument.arkiv.consumer.BidragOrganisasjonConsumer;
import no.nav.bidrag.dokument.arkiv.dto.Saksbehandler;
import no.nav.bidrag.dokument.arkiv.utils.TokenUtils;
import org.springframework.stereotype.Service;

@Service
public class SaksbehandlerInfoManager {

  private final BidragOrganisasjonConsumer bidragOrganisasjonConsumer;
  private final OidcTokenManager oidcTokenManager;

  public SaksbehandlerInfoManager(BidragOrganisasjonConsumer bidragOrganisasjonConsumer,
      OidcTokenManager oidcTokenManager) {
    this.bidragOrganisasjonConsumer = bidragOrganisasjonConsumer;
    this.oidcTokenManager = oidcTokenManager;
  }


  public String hentSaksbehandlerBrukerId(){
    try {
      return TokenUtils.henteSubject(oidcTokenManager.fetchTokenAsString());
    } catch (Exception e){
      return null;
    }
  }
  public Optional<Saksbehandler> hentSaksbehandler(){
    try {
      var saksbehandlerIdent = hentSaksbehandlerBrukerId();
      if (saksbehandlerIdent == null){
        return Optional.empty();
      }
      var saksbehandlerNavn = bidragOrganisasjonConsumer.hentSaksbehandlerInfo(saksbehandlerIdent).getNavn();
      return Optional.of(new Saksbehandler(saksbehandlerIdent, saksbehandlerNavn));
    } catch (Exception e){
      return Optional.empty();
    }
  }

}

File: src/main/java/no/nav/bidrag/dokument/arkiv/security/SaksbehandlerInfoManager.kt
package no.nav.bidrag.dokument.arkiv.security

import no.nav.bidrag.commons.security.service.OidcTokenManager
import no.nav.bidrag.dokument.arkiv.consumer.BidragOrganisasjonConsumer
import no.nav.bidrag.dokument.arkiv.dto.Saksbehandler
import no.nav.bidrag.dokument.arkiv.utils.TokenUtils.henteSubject
import org.springframework.stereotype.Service

@Service
class SaksbehandlerInfoManager(
    private val bidragOrganisasjonConsumer: BidragOrganisasjonConsumer,
    private val oidcTokenManager: OidcTokenManager
) {
    fun hentSaksbehandlerBrukerId(): String? {
        return try {
            henteSubject(oidcTokenManager.fetchTokenAsString())
        } catch (e: Exception) {
            null
        }
    }

    fun hentSaksbehandler(_saksbehandlerIdent: String? = null): Saksbehandler? {
        return try {
            val saksbehandlerIdent = _saksbehandlerIdent ?: hentSaksbehandlerBrukerId() ?: return null
            val saksbehandlerNavn = bidragOrganisasjonConsumer.hentSaksbehandlerInfo(saksbehandlerIdent).navn?.verdi
            Saksbehandler(saksbehandlerIdent, saksbehandlerNavn)
        } catch (e: Exception) {
            null
        }
    }
}
